sidechain/storage/src/interface.rs

```diff
@@ -65,6 +65,18 @@ pub trait FetchBlocks<SignedBlock: SignedBlockT> {
 		block_hash: &BlockHash,
 		shard_identifier: &ShardIdentifierFor<SignedBlock>,
 	) -> Result<Vec<SignedBlock>>;
+
+	/// Fetch all blocks within a range, defined by a starting block (lower bound) and end block (upper bound) hash.
+	///
+	/// Does NOT include the bound defining blocks in the result. ]from..until[.
+	/// Returns an empty vector if 'from' cannot be found in storage.
+	/// Returns the same as 'fetch_all_blocks_after' if 'until' cannot be found in storage.
+	fn fetch_blocks_in_range(
+		&self,
+		block_hash_from: &BlockHash,
+		block_hash_until: &BlockHash,
+		shard_identifier: &ShardIdentifierFor<SignedBlock>,
+	) -> Result<Vec<SignedBlock>>;
```

Comment:
Storage has a new method in the 'fetch' interface: Fetch blocks in a range, specified by start and end block hash.